File: src/main/java/be/bluexin/expanse/worker/example/UpdaterHp.java
package be.bluexin.expanse.worker.example;

import be.bluexin.expanse.worker.Component;
import be.bluexin.expanse.worker.EntityStore;
import be.bluexin.expanse.worker.Health;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Random;
import java.util.concurrent.ExecutionException;

public class UpdaterHp {

    static Component<Health> HEALTH_JAVA = new Component<>("HEALTH_JAVA", 2, Health.class);

    public static void main(String[] args) throws InterruptedException, ExecutionException {
        Logger logger = LoggerFactory.getLogger("HP Updater");
        logger.info("Starting connection");
        EntityStore.INSTANCE.connectAeronFuture().get();
        logger.info("Entering stuff");

        Random rng = new Random();
        for (int i = 0; i < 20; i++) {
            Health hp = new Health(rng.nextFloat(), rng.nextFloat());
            EntityStore.INSTANCE.get(2).set(HEALTH_JAVA, hp);

            logger.info("Set hp to " + hp);
            Thread.sleep(1000);
        }

        Thread.sleep(3000);
        EntityStore.INSTANCE.disconnectAeronBlocking();
    }
}

File: src/main/kotlin/be/bluexin/expanse/worker/example/updaterHp.kt
package be.bluexin.expanse.worker.example

import be.bluexin.expanse.worker.Component
import be.bluexin.expanse.worker.EntityStore
import be.bluexin.expanse.worker.Health
import be.bluexin.expanse.worker.Position
import kotlinx.coroutines.ExperimentalCoroutinesApi
import kotlinx.coroutines.delay
import kotlinx.coroutines.runBlocking
import mu.KotlinLogging
import kotlin.random.Random

@UseExperimental(ExperimentalCoroutinesApi::class)
fun main() {
    runBlocking {
        val logger = KotlinLogging.logger("main")
        logger.info { "Starting connection" }
        EntityStore.connectAeron()
        logger.info { "Entering stuff" }

        val rng = Random(Random.nextLong())
        for (i in 0 until 20) {
            val hp = Health(rng.nextFloat(), rng.nextFloat())
            // We store health in entity with ID 2
            EntityStore[2][Component.HEALTH] = hp

            logger.info { "Set hp to: $hp" }
            delay(1000)
        }

        delay(3000)
        EntityStore.disconnectAeron()
    }

}
